### ISSUE TITLE
Update failed: can't find the current version. Please try again later

### ISSUE THREAD

**Issue by serge-sb** [ytdl-org/youtube-dl#26168]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2020.07.28**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
Traceback (most recent call last):
  File "C:\Python\Python34\lib\urllib\request.py", line 1183, in do_open
  File "C:\Python\Python34\lib\http\client.py", line 1137, in request
  File "C:\Python\Python34\lib\http\client.py", line 1182, in _send_request
  File "C:\Python\Python34\lib\http\client.py", line 1133, in endheaders
  File "C:\Python\Python34\lib\http\client.py", line 963, in _send_output
  File "C:\Python\Python34\lib\http\client.py", line 898, in send
  File "C:\Python\Python34\lib\http\client.py", line 1287, in connect
  File "C:\Python\Python34\lib\ssl.py", line 362, in wrap_socket
  File "C:\Python\Python34\lib\ssl.py", line 580, in __init__
  File "C:\Python\Python34\lib\ssl.py", line 807, in do_handshake
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpq2whuxta\build\youtube_dl\update.py", line 46, in update_self
  File "C:\Python\Python34\lib\urllib\request.py", line 464, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 482, in _open
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpq2whuxta\build\youtube_dl\utils.py", line 2736, in https_open
  File "C:\Python\Python34\lib\urllib\request.py", line 1185, in do_open
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)>

ERROR: can't find the current version. Please try again later.
```

## Description

Windows 7 64bit
Just running "youtube-dl.exe -U" as usual, bit this time it shows this. I've downloaded the latest version manually from https://yt-dl.org/latest/youtube-dl.exe and the problem still persists. Maybe exe-file needs to be rebuild with updated python?..

**Comment by serge-sb** [comment]
@remitamine Could you please review the updated issue? Sorry, I have truncated a bit verbose log output at first, but now it is complete. Please let me know about the ticket that is duplicate to this, as I could now find any. Also note, that other issues about update failures relate to non-Windows OSes, this is really important, as Windows version contains embedded Python.

**Comment by zrneely** [comment]
For anyone else who comes here and finds this issue, it appears this is being tracked in https://github.com/ytdl-org/youtube-dl/issues/25541.

**Comment by serge-sb** [comment]
@zrneely That issue is locked. I did a lot of searching about this issue here and finally I found #25622 where @dstftw says, that the problem with updating to a newer version of python is that py2exe supports 3.4 at most. I have found this branch of one of the py2exe forks, that seems to be able to support 3.8: https://github.com/albertosottile/py2exe/tree/py38
Hope to receive any feedback about this info. Thanks!
